src/python/pants/init/plugin_resolver.py

```diff
@@ -88,11 +107,7 @@ def __init__(
         interpreter_constraints: Optional[InterpreterConstraints] = None,
```

Comment:
If my read is correct, this is only used for testing? If so, I think we could remove to make this PR's change even more clear: no matter what, use sys.executable.

Reply:
There are tests that validate fingerprint changes with different interpreters: will leave for now.